proof_of_play.go

```diff
@@ -158,3 +168,58 @@ func (p *proofOfPlay) expire(ad Ad) error {
 		return false, nil
 	})
 }
+
+func isLeaseExpired(ad Ad) bool {
+	exp, ok := ad["lease_expiry"].(float64)
+	if !ok {
+		return true
+	}
+
+	expiry := int64(exp)
+	now := int64(time.Now().Unix())
+	return now >= expiry
+}
+
+func (p *proofOfPlay) retryFailedPoPs() {
```

Comment:
I think this can be simplified. For starters, I'd just use a ticker instead of JavaScript style setTimeout calls. Something like:

ticker := time.NewTicker(...)
for range ticker.C {
   // go over failed pops
}

And for the actual failed pop store you can use a channel. On each tick, you'll simply read from this channel, check if it's expired, if not write it back to the request channel.

Reply:
Done.